lib/cloudinary/carrier_wave/remote.rb

```diff
@@ -4,8 +4,22 @@ def download!(uri, *args)
     if respond_to?(:process_uri)
       uri = process_uri(uri)
     else # Backward compatibility with old CarrierWave
-      remote_url_unsafe_chars = /([^a-zA-Z0-9_.\-\/:?&=]+)/ # In addition allow query string characters: "?","&" and "="
-      uri = URI.parse(Cloudinary::Utils.smart_escape(Cloudinary::Utils.smart_unescape(uri), remote_url_unsafe_chars))
+      remote_url_unsafe_chars = /([^a-zA-Z0-9_.\-\/:?&=]+)/ # In addition allow query string characters: "?","&" and "="    
+      smart_escaped_uri = URI.parse(Cloudinary::Utils.smart_escape(Cloudinary::Utils.smart_unescape(uri), remote_url_unsafe_chars))
+      
+      uri_array = smart_escaped_uri.to_s.split("/")
+      https_count = 0
+      uri_array.each do |word|
+        if word == "https:"
+          https_count += 1
+        end
+      end
+      
+      if https_count>1
+        uri = URI.parse(uri)

```

Comment:
Another way to do code like this in Ruby would be:
```ruby
uri =
  if https_count > 1
    URI.parse(uri)
  else
    smart_escaped_uri
  end
```

Or you could use the ternary operator:
```ruby
uri = https_count > 1 ? URI.parse(uri) : smart_escaped_uri
```

You could do without the intermediate variable smart_escaped_uri at all. Just keep setting uri at first and the reset it later on after a test - this goes against concepts like immutability but this is Ruby:
```ruby
uri = URI.parse(uri) if https_count > 1
```

And then thinking about what is https_count? You're testing for an expression that could be done with a regex...
```ruby
# either we test for this specific client host
dont_be_smart_about_this = /\/https%3A%2F%2Fphotos.watchmedier.dk%2F[a-zA-Z0-9%\.]+\.(jpg|png|gif)$/
# or we choose to be more generic about it
dont_be_smart_about_this = /\/https%3A%2F%2F[a-zA-Z0-9%\.]+\.(jpg|png|gif)$/
# and maybe we even don't want to limit ourselves to possible extensions
dont_be_smart_about_this = /\/https%3A%2F%2F[a-zA-Z0-9%\.]$/

uri = URI.parse(uri) if uri.match? dont_be_smart_about_this
```

Reply:
Thanks for the remark! I forgot how fun ruby can be for writing. Changed the uri assignment to a one liner.

We want to avoid smart unescaping and then escaping only for uris that contain https twice (or more).
Also I am checking the smart escaped string, since in the original uri the strings are different - "https://" the first time and "https%3A%2F%2F" the second time.

What I think fits best is to check for https:// + ay text {at least twice}:  `/https:\/\/[\w+]{2}/`